src/System.Collections/tests/Generic/CollectionExtensionsTests.cs

```diff
@@ -72,5 +72,66 @@ public void GetValueOrDefault_NullIReadOnlyDictionary_ThrowsArgumentNullExceptio
             Assert.Throws<ArgumentNullException>("dictionary", () => dictionary.GetValueOrDefault("key"));
             Assert.Throws<ArgumentNullException>("dictionary", () => dictionary.GetValueOrDefault("key", "value"));
         }
+
+        [Fact]
+        public void TryAdd_NullIDictionary_ThrowsArgumentNullException()
+        {
+            IDictionary<string, string> dictionary = null;
+            Assert.Throws<ArgumentNullException>("dictionary", () => dictionary.TryAdd("key", "value"));
+        }
+
+        [Fact]
+        public void TryAdd_NullKeyIDictionary_ThrowsArgumentNullException()
+        {
+            IDictionary<string, string> dictionary = new SortedDictionary<string, string>();
+            Assert.Throws<ArgumentNullException>("key", () => dictionary.TryAdd(null, "value"));
+        }
+
+        [Fact]
+        public void TryAdd_KeyDoesntExistInIDictionary_ReturnsTrue()
+        {
+            IDictionary<string, string> dictionary = new SortedDictionary<string, string>();
+            Assert.True(dictionary.TryAdd("key", "value"));
+            Assert.Equal("value", dictionary["key"]);
+        }
+
+        [Fact]
+        public void TryAdd_KeyExistsInIDictionary_ReturnsFalse()
+        {
+            IDictionary<string, string> dictionary = new SortedDictionary<string, string>() { ["key"] = "value" };
+            Assert.False(dictionary.TryAdd("key", "value2"));
+            Assert.Equal("value", dictionary["key"]);
+        }
+
+        [Fact]
+        public void Remove_NullIDictionary_ThrowsArgumentNullException()
+        {
+            IDictionary<string, string> dictionary = null;
+            Assert.Throws<ArgumentNullException>("dictionary", () => dictionary.Remove("key", out var value));
```

Comment:
We could assign a value to the out parameter before calling Remove and test that the value did not changed when an exception is thrown.